raft.c

```diff
@@ -2108,6 +2146,28 @@ void handleShardGroupGet(RedisRaftCtx *rr, RaftReq *req)
     RaftReqFree(req);
 }
 
+static void handleNodeRemoved(RedisRaftCtx *rr, RaftReq *req)
+{
+    if (req->r.node_removed.id != raft_get_nodeid(rr->raft)) {
+        LOG_ERROR("Received invalid noderemoved message with id : %d.",
+                  req->r.node_removed.id);
+        return;
+    }
```

Comment:
I'm not against this code, but if this happens, we probably have other issues too.

Reply:
Just it is a precaution but I think leader may send this message on replay or if another leader is elected and that leader applies this entry after becoming the leader.  If we remove a node and start another node at the same machine (ip:port), new node would shutdown if I'm not mistaken